autopush/endpoint.py

```diff
@@ -360,36 +366,56 @@ def put(self, token):
         self.start_time = time.time()
         fernet = self.ap_settings.fernet
 
-        version, data = parse_request_params(self.request)
-        if data and len(data) > self.ap_settings.max_data:
-            self.set_status(401)
-            log.msg("Data too large", **self._client_info())
-            self.write("Data too large")
-            return self.finish()
-
         d = deferToThread(fernet.decrypt, token.encode('utf8'))
-        d.addCallback(self._token_valid, version, data)
+        d.addCallback(self._token_valid)
         d.addErrback(self._token_err)
         d.addErrback(self._response_err)
+    post = put
 
     #############################################################
     #                    Callbacks
     #############################################################
-    def _token_valid(self, result, version, data):
+    def _token_valid(self, result):
         """Called after the token is decrypted successfully"""
-        self.uaid, chid = result.split(":")
-        notification = Notification(version=version, data=data,
-                                    channel_id=chid)
+        self.uaid, self.chid = result.split(":")
         d = deferToThread(self.ap_settings.router.get_uaid, self.uaid)
-        d.addCallback(self._uaid_lookup_results, notification)
+        d.addCallback(self._uaid_lookup_results)
         d.addErrback(self._uaid_not_found_err)
         self._db_error_handling(d)
 
-    def _uaid_lookup_results(self, result, notification):
+    def _uaid_lookup_results(self, result):
         """Process the result of the AWS UAID lookup"""
         # Save the whole record
         router_key = result.get("router_type", "simplepush")
         router = self.ap_settings.routers[router_key]
+
+        # Only simplepush uses version/data out of body/query, GCM/APNS will
+        # use data out of the request body 'WebPush' style.
+        if router_key == "simplepush":
+            version, data = parse_request_params(self.request)
+        else:
+            # We need crypto headers
```

Comment:
This is only for Web Push, right? Or are we planning to send encrypted messages via GCM/APNS, too?